lib/ecto_diff.ex

```diff
@@ -276,10 +279,14 @@ defmodule EctoDiff do
     }
   end
 
-  defp fields(%struct{} = previous, %struct{} = current) do
+  defp fields(%struct{} = previous, %struct{} = current, opts) do
+    include_virtual_fields? = Keyword.get(opts, :include_virtual_fields, false)
     field_names = struct.__schema__(:fields) -- struct.__schema__(:embeds)
 
     field_names
+    |> then(fn names ->
+      if include_virtual_fields?, do: names ++ struct.__schema__(:virtual_fields), else: field_names
+    end)
```

Comment:
:information_source: Instead of using `then` we can make a private function:
```
  defp maybe_include_virtual_fields?(names, struct, true), do: names ++ struct.__schema__(:virtual_fields)
  defp maybe_include_virtual_fields?(names, _struct, false), do: names
```